app/controllers/case_workers/admin/management_information_controller.rb

```diff
@@ -13,7 +13,14 @@ def index
 
   def download
     record = Stats::StatsReport.most_recent_by_type(params[:report_type])
-    send_data record.report, filename: record.download_filename, type: 'text/csv'
+    if record.document?
+      data = open(record.document_url).read
+      content_type = record.document_content_type
+    else
+      data = record.report
```

Comment:
**NOTE:** We will give a transition period for the new reports to be generated and stored in S3. Once all of them are using this new approach, we should be able to remove the old way of storing things.